src/AzureAppConfigurationImpl.ts

```diff
@@ -195,17 +223,58 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             keyValues.push([key, value]);
         }
 
-        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#clearLoadedKeyValues(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.#configMap.set(k, v);
         }
     }
 
+    async #clearLoadedKeyValues() {
+        for(const key of this.#configMap.keys()) {
+            if (key !== featureManagementKeyName) {
+                this.#configMap.delete(key);
+            }
+        }
+    }
+
+    async #loadFeatureFlags() {
+        const featureFlags: FeatureFlagValue[] = [];
+        const featureFlagSelectors = getValidSelectors(this.#options?.featureFlagOptions?.selectors);
+        for (const selector of featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter
+            };
+            if (this.#requestTracingEnabled) {
+                listOptions.requestOptions = {
+                    customHeaders: {
+                        [CorrelationContextHeaderName]: createCorrelationContextHeader(this.#options, this.#isInitialLoadCompleted)
+                    }
+                }
+            }
+
+            const settings = this.#client.listConfigurationSettings(listOptions);

```

Comment:
Did JS SDK release collection monitoring API? I see they had a PR for this but not sure if this exists in the latest stable release: https://github.com/Azure/azure-sdk-for-js/pull/28555

Reply:
The PR is in sdk v1.6.0. Later I'll create another PR to update the dependencies and support the page etag.

Reply:
Is this version released now? I can see the public docs for the new API https://learn.microsoft.com/en-us/javascript/api/@azure/app-configuration/listconfigurationsettingpage?view=azure-node-latest